src/quo2/components/info/page_nav.cljs

```diff
@@ -0,0 +1,219 @@
+                                             :weight :font-medium
+                                             :style  {:padding-right 4
+                                                      :margin-left   2
+                                                      :text-align    :left
+                                                      :line-height   18
+                                                      :color         text-secondary-color}}
+                                    mid-section-description]])]])]))
+
+(defn right-section-icon
+  [props {:keys [bg icon icon-color]}]
+  [rn/view {:style (merge
+                    icon-styles
+                    {:background-color bg
+                     :width            32
+                     :height           32}
+                    props)}
+   [icons/icon icon (icon-props icon-color :big)]])
+
+(defn page-nav
+  [{:keys [one-icon-align-left? horizontal-description? align-mid page-nav-color page-nav-background-uri mid-section-type mid-section-icon mid-section-main-text mid-section-left-icon
+           mid-section-right-icon mid-section-description mid-section-description-color mid-section-description-icon mid-section-description-user-icon
+           mid-section-main-text-icon-color left-section-icon left-section-icon-color
+           left-section-icon-bg-color right-section-icons]
+    :or   {align-mid                         false
+           page-nav-color                    :transparent
+           horizontal-description?           false
+           one-icon-align-left?              false
+           page-nav-background-uri           ""
+           mid-section-type                  :text-with-description
+           mid-section-icon                  :wallet
+           mid-section-main-text             "# general"
+           mid-section-left-icon             :wallet
+           mid-section-right-icon            :wallet
+           mid-section-description           "Description"
+           mid-section-description-color     "black"
+           mid-section-description-icon      :placeholder20
+           mid-section-description-user-icon "https://i.picsum.photos/id/810/200/300.jpg?hmac=HgwlXd-OaLOAqhGyCiZDUb_75EgUI4u0GtS7nfgxd8s"
+           mid-section-main-text-icon-color  "nil"
+           left-section-icon                 :peach20
+           left-section-icon-color           "none"
+           left-section-icon-bg-color        colors/neutral-30
+           right-section-icons               [{:bg       colors/primary-40
+                                               :no-color true
+                                               :icon     :main-icons/placeholder20}]}}]
+  (let [{:keys [height width]}      (dimensions/window)
+        put-middle-section-on-left? (or (true? align-mid)
+                                        (> (count right-section-icons) 1))
+        mid-section-props {:mid-section-type                 mid-section-type
```

Comment:
can we just pass `mid-section-props` map as a parameter to page-nav ?

Reply:
Trying it out